File: src/main/java/cn/qiuye/gtmoremachine/data/recipe/GTMMRecipes.java
package cn.qiuye.gtmoremachine.data.recipe;

import cn.qiuye.gtmoremachine.data.recipe.misc.Assembler;
import cn.qiuye.gtmoremachine.data.recipe.misc.Vanilla;

import net.minecraft.data.recipes.FinishedRecipe;

import java.util.function.Consumer;

public class GTMMRecipes {

    public static void init(Consumer<FinishedRecipe> provider) {
        Vanilla.init(provider);
        Assembler.init(provider);
    }
}

File: src/main/java/cn/qiuye/gtmoremachine/data/recipes/GTMMRecipes.kt
package cn.qiuye.gtmoremachine.data.recipes

import cn.qiuye.gtmoremachine.GTmm
import cn.qiuye.gtmoremachine.data.recipes.misc.AE2AssemblerRecipe
import cn.qiuye.gtmoremachine.data.recipes.misc.AssemblerRecipe
import cn.qiuye.gtmoremachine.data.recipes.misc.VanillaRecipe

import net.minecraft.data.recipes.RecipeOutput

object GTMMRecipes {

	@JvmStatic
	fun init(provider: RecipeOutput) {
		if (GTmm.Mods.isAE2Loaded()) {
			AE2AssemblerRecipe.init(provider)
		}
		AssemblerRecipe.init(provider)
		VanillaRecipe.init(provider)
	}
}
